packages/bus/src/RetryStrategies/ExponentialBackoffRetryStrategy.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { ExponentialBackoffRetryStrategy } from './ExponentialBackoffRetryStrategy';
+
+describe('ExponentialBackoffRetryStrategy', () => {
+  it('should not retry if function does not throw error', async () => {
+    // arrange
+    const retryStrategy = new ExponentialBackoffRetryStrategy({ maxDepth: 1 });
+    const input = jest.fn().mockResolvedValue(undefined);
+
+    // act
+    await retryStrategy.acquire(input);
+
+    // assert
+    expect(input).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return a result execution immediately', async () => {
+    const retryStrategy = new ExponentialBackoffRetryStrategy({ maxDepth: 1 });
+    const input = jest.fn().mockReturnValue(undefined);
+
+    await retryStrategy.acquire(input);
+
+    expect(input).toHaveBeenCalledTimes(1);
+  });
+
+  it('should prevent retries if error does not have a correct code', async () => {
+    // arrange
+    const retryStrategy = new ExponentialBackoffRetryStrategy({ maxDepth: 1 });
+    const input = jest.fn().mockRejectedValue(new Error('Unhandled error'));
+
+    // act
+    const result = retryStrategy.acquire(input);
+
+    // assert
+    await expect(result).rejects.toThrow('Unhandled error');
+    expect(input).toHaveBeenCalledTimes(1);
+  });
+
+  it('should retry 2 times and throw an error', async () => {
+    const retryStrategy = new ExponentialBackoffRetryStrategy({ maxDepth: 2 });
+    const error = new Error('Unhandled error');
+    (error as any).code = 'ECONNRESET';
+    const input = jest.fn().mockRejectedValue(error);
+
+    const result = retryStrategy.acquire(input);
+
+    await expect(result).rejects.toThrow(error);
+    expect(input).toHaveBeenCalledTimes(3);
+  });
+
+  it('should return a result execution after a few retries', async () => {
```

Comment:
Just look at the test case title: "should return a result". You should define expectations accordingly.